### ISSUE TITLE
cannot open the activity app after editing/opening a mindmap

### ISSUE THREAD

**Issue by szaimen** [ACTom/files_mindmap#99]
I am currently on v20, NC 18.0.2, Ubuntu 18.04, PHP 7.2

After opening/editing a mindmap I cannot open the activity app with this user.

To reproduce:
0. install mindmap v20
1. open/edit a mindmap
2. close and save
3. try to open the activity app
4. See a never ending spinning wheel

expected:
I shouldn't see this spinning wheel but instead the activity stream.

These errors are in the logs:

[PHP] Error: copy(/var/www/nextcloud/core/img/filetypes/mindmap.svg): failed to open stream: Permission denied at /var/www/nextcloud/apps/files_mindmap/lib/Migration/InstallStep.php#41

POST /settings/apps/enable
from 87.123.148.145 by admin at 2020-03-17T21:26:25+01:00

[PHP] Error: file_put_contents(/var/www/nextcloud/core/js/mimetypelist.js): failed to open stream: Permission denied at /var/www/nextcloud/core/Command/Maintenance/Mimetype/UpdateJS.php#57

POST /settings/apps/enable
from 87.123.148.145 by admin at 2020-03-17T21:26:25+01:00

[no app in context] Error: Exception: substr() expects parameter 3 to be integer, boolean given at <<closure>>

0. /var/www/nextcloud/lib/private/AppFramework/App.php line 125
   OC\AppFramework\Http\Dispatcher->dispatch(OCA\Activity\Controller\APIv2 {}, "getFilter")
1. /var/www/nextcloud/lib/private/AppFramework/Routing/RouteActionHandler.php line 47
   OC\AppFramework\App::main("APIv2Controller", "getFilter", OC\AppFramework\ ... {}, {filter: "all",_ ... "})
2. <<closure>>
   OC\AppFramework\Routing\RouteActionHandler->__invoke({filter: "all",_ ... "})
3. /var/www/nextcloud/lib/private/Route/Router.php line 299
   call_user_func(OC\AppFramework\ ... {}, {filter: "all",_ ... "})
4. /var/www/nextcloud/ocs/v1.php line 82
   OC\Route\Router->match("/ocsapp/apps/ac ... l")
5. /var/www/nextcloud/ocs/v2.php line 24
   require_once("/var/www/nextcloud/ocs/v1.php")

GET /ocs/v2.php/apps/activity/api/v2/activity/all?format=json&previews=true&since=0
from 87.123.148.145 by admin at 2020-03-17T21:28:06+01:00

**Comment by ACTom** [comment]
Judging from the error message, it is because your core directory does not have write permission during installation. When the app is installed, you need to write files to the core directory. You can try the following operations:
1.Copy https://raw.githubusercontent.com/ACTom/files_mindmap/master/img/mindmap.svg to /var/www/nextcloud/core/img/filetypes/mindmap.svg
2. Manually execute in the installation directory
`occ maintenance: mimetype: update-js`
`occ files: scan`

**Comment by szaimen** [comment]
Okay thanks, I will try that later on. 
Is that the reason why you want the icon to become part of the core?

**Comment by szaimen** [comment]
The solution worked for me but I think that will only be solved if the icon is part of the server.

**Comment by ACTom** [comment]
> 
> 
> Okay thanks, I will try that later on.
> Is that the reason why you want the icon to become part of the core?

Yes, I think there is a problem writing files to the core directory during installation

**Comment by ACTom** [comment]
I submitted two PRs in nextcloud / server, this problem will be completely solved in Nextcloud 19.0.
